src/App.tsx

```diff
@@ -1,10 +1,91 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { UserWarning } from './UserWarning';
+import { Footer } from './components/footer';
+import { Main } from './components/main';
+import { ErrorNotification } from './components/errorNotification';
+import { Header } from './components/header';
+import { Todo } from './types/Todo';
+import { SelectOptions } from './types/SelectOptions';
+import { getTodos } from './api/todos';
 
-const USER_ID = 0;
+const USER_ID = 6447;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [onError, setOnError] = useState(true);
+  const [visibleTodos, setVisibleTodos] = useState<Todo[]>([]);
+  const [completedTodosId, setCompletedTodosId] = useState<number[]>([]);
+
+  const filteredCompletedTodosId = (allTodos:Todo[]) => {
+    return allTodos
+      .filter((todo: Todo) => todo.completed)
+      .map(todoItem => todoItem.id);
+  };
+
+  const filterTodosBySelectOptions = (type: SelectOptions) => {
+    switch (type) {
+      case SelectOptions.ACTIVE:
+        setVisibleTodos(todos.filter((todo: Todo) => !todo.completed));
+        break;
+      case SelectOptions.COMPLETED:
+        setVisibleTodos(todos.filter((todo: Todo) => todo.completed));
+        break;
+      case SelectOptions.ALL:
+      default:
+        setVisibleTodos(todos);
+    }
+  };
+
+  const loadTodosFromServer = async () => {
+    try {
+      const todosData = await getTodos(USER_ID);
+
+      if (!todosData) {
+        setOnError(true);
+      } else {
+        setTodos(todosData);
+        setVisibleTodos(todosData);
+        setCompletedTodosId(filteredCompletedTodosId(todosData));
+      }
+    } catch (err) {
+      if (err instanceof Error) {
+        // eslint-disable-next-line no-alert
+        alert(err);
+      }
+    }
+  };
+
+  useEffect(() => {
+    loadTodosFromServer();
+  }, []);
+
+  useEffect(() => {
+    const timeoutID = setTimeout(() => {
+      setOnError(false);
+    }, 3000);
+
+    return () => clearTimeout(timeoutID);
+  }, []);
```

Comment:
small advice for the future. if you want to hide error after 3 seconds you have to add `onError` as a dependency and if it's true set timeout to hide.